### ISSUE TITLE
`brew update` reports spurious change for files moved within tap repo

### ISSUE THREAD

**Issue by apjanke** [Homebrew/brew#87]
If a formula file in a tap is moved between the allowable formula directories (`/`, `/Formula`, and `/HomebrewFormula`), it will be reported as changed by the next `brew update` that picks it up.

The reported change will look something like this.

```
$ brew update
Updated 1 tap (homebrew/versions).
==> New Formulae
homebrew/versions/allegro4
homebrew/versions/android-ndk-r10d
homebrew/versions/android-ndk-r10e
homebrew/versions/ansible14
homebrew/versions/ansible18
```

This blocks rearranging the standard taps to move formula files from the root dir to `Formula`, which is our current preferred organization.

Blocks homebrew/homebrew-boneyard#108.

**Comment by xu-cheng** [comment]
I think a more serious blocker is the `FormulaVersions` won't work after the formula directory change. As the result, we may get wrong revision number for bottles.

**Comment by mikemcquaid** [comment]
> I think a more serious blocker is the FormulaVersions won't work after the formula directory change. As the result, we may get wrong revision number for bottles.

Yup, we'll need to make sure all these sort of things work after renaming (they can be made to)

**Comment by UniqMartin** [comment]
The `update-report`-related issue of reporting spurious changes has been addressed in #480.

**Comment by MikeMcQuaid** [comment]
> The update-report-related issue of reporting spurious changes has been addressed in #480.

Sounds sufficient to me for now at least.

**Comment by apjanke** [comment]
I tested this out with a dummy repo, moving a bunch of formulae from the root dir to `Formula/`, and it worked – no spurious messages. Looks like it's safe to reorganize taps now.
